migrate_cpp/rewriter.h

```diff
@@ -0,0 +1,210 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_MIGRATE_CPP_REWRITER_H_
+#define CARBON_MIGRATE_CPP_REWRITER_H_
+
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/ASTTypeTraits.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendAction.h"
+#include "llvm/ADT/DenseMap.h"
+#include "migrate_cpp/output_segment.h"
+
+namespace Carbon {
+
+// `OutputWriter` is ressponsible for traversing the tree of `OutputSegment`s
+// and writing the correct data to its member `output`.
+class OutputWriter {
```

Comment:
For declaration order on this class, I'd typically expect public first, private second (and regardless, `private` explicitly marked).
https://google.github.io/styleguide/cppguide.html#Declaration_Order

Also, noting the public class members: can those be shifted private?
https://google.github.io/styleguide/cppguide.html#Access_Control

Note, I think these may be linked. If the data members were private, then I think this class could comfortable make the structs private, and it would be easier to find the API. If it's intended to behave more as a `struct`, which it kind of is now with one operative function, then I wouldn't expect it to have private members; and I don't know how much the class benefits from having these private. But I think I do think it'd be better to lean more towards one approach or the other (`struct OutputWriter` with only public members, or `class OutputWriter` with a more clearly defined API), from a style perspective.

Reply:
Done.